eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java

```diff
@@ -78,45 +77,73 @@ public final class BinarizeMojo extends SafeMojo {
-                Logger.info(this, String.format("Building rust project.."));
-                final Process building = new ProcessBuilder("cargo", "build")
-                    .directory(file)
-                    .start();
-                try {
-                    building.waitFor();
-                } catch (final InterruptedException exception) {
-                    Thread.currentThread().interrupt();
-                    throw new BuildFailureException(
-                        String.format(
-                            "Interrupted while building %s",
-                            file
-                        ),
-                        exception
-                    );
-                }
-                if (building.exitValue() != 0) {
-                    Logger.error(this, "There was an error in compilation");
-                    try (VerboseProcess process = new VerboseProcess(building)) {
-                        new Unchecked<>(
-                            new LengthOf(
-                                new TeeInput(
-                                    new InputOf(process.stdoutQuietly()),
-                                    new OutputTo(new ByteArrayOutputStream())
-                                )
-                            )
-                        ).value();
-                    }
-                    throw new BuildFailureException(
-                        String.format(
-                            "Failed to build cargo project with dest = %s",
-                            file
-                        )
-                    );
-                }
+        for (final File project: targetDir.toPath().resolve("Lib").toFile().listFiles()) {
+            if (project.isDirectory() && project.toPath().resolve("Cargo.toml").toFile().exists()) {
+                this.build(project);
             }
         }
     }
 
+    /**
+     * Builds cargo project.
+     * @param project Path to the project.
+     * @throws IOException If any issues with IO.
+     */
+    private void build(final File project) throws IOException {
+        final File target = project.toPath().resolve("target").toFile();
+        final File cached = cache
+            .resolve("Lib")
+            .resolve(project.getName())
+            .resolve("target").toFile();
+        if (cached.exists()) {
+            Logger.info(
+                this,
+                String.format(
+                    "Copying %s to %s",
+                    cached,
+                    target
+                )
+            );
+            FileUtils.copyDirectory(
+                cached,
+                target
+            );
+        }
+        Logger.info(this, "Building rust project..");
+        try (
+            VerboseProcess proc = new VerboseProcess(
+                new ProcessBuilder("cargo", "build")
+                    .directory(project)
+            )
+        ) {
+            proc.stdout();
```

Comment:
@levBagryansky We can use `stdoutQuietly()` and avoid `IllegalArgumentException` if something was wrong with the build. But if that behaviour is intentional, that is ok.

Reply:
@Graur yes, this is intentional.